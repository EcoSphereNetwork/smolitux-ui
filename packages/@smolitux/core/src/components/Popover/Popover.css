.smolitux-popover {
  position: absolute;
  background-color: var(--popover-bg, white);
  border-radius: 0.375rem;
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
  z-index: var(--popover-z-index, 1000);
  max-width: var(--popover-max-width, none);
  width: max-content;
  border: 1px solid var(--popover-border-color, #e5e7eb);
  opacity: 0;
  visibility: hidden;
  transition: opacity 0.2s ease, visibility 0.2s ease;
}

.smolitux-popover--visible {
  opacity: 1;
  visibility: visible;
}

/* Popover Header */
.smolitux-popover-header {
  padding: 0.5rem 1rem;
  border-bottom: 1px solid var(--popover-border-color, #e5e7eb);
  font-weight: 600;
  color: var(--popover-header-color, #111827);
}

/* Popover Content */
.smolitux-popover-content {
  padding: 0.5rem 1rem;
  color: var(--popover-content-color, #374151);
}

/* Popover Arrow */
.smolitux-popover-arrow {
  position: absolute;
  width: 0.75rem;
  height: 0.75rem;
  background-color: var(--popover-bg, white);
  border: 1px solid var(--popover-border-color, #e5e7eb);
  transform: rotate(45deg);
  z-index: -1;
}

/* Arrow Positions */
.smolitux-popover--top .smolitux-popover-arrow,
.smolitux-popover--top-start .smolitux-popover-arrow,
.smolitux-popover--top-end .smolitux-popover-arrow {
  bottom: -0.375rem;
  border-top: none;
  border-left: none;
}

.smolitux-popover--right .smolitux-popover-arrow,
.smolitux-popover--right-start .smolitux-popover-arrow,
.smolitux-popover--right-end .smolitux-popover-arrow {
  left: -0.375rem;
  border-right: none;
  border-bottom: none;
}

.smolitux-popover--bottom .smolitux-popover-arrow,
.smolitux-popover--bottom-start .smolitux-popover-arrow,
.smolitux-popover--bottom-end .smolitux-popover-arrow {
  top: -0.375rem;
  border-bottom: none;
  border-right: none;
}

.smolitux-popover--left .smolitux-popover-arrow,
.smolitux-popover--left-start .smolitux-popover-arrow,
.smolitux-popover--left-end .smolitux-popover-arrow {
  right: -0.375rem;
  border-left: none;
  border-top: none;
}

/* Animations */
@keyframes smolitux-popover-fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes smolitux-popover-fade-out {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}

.smolitux-popover--animating-in {
  animation: smolitux-popover-fade-in 0.2s ease forwards;
}

.smolitux-popover--animating-out {
  animation: smolitux-popover-fade-out 0.2s ease forwards;
}

/* Trigger Element */
.smolitux-popover-trigger {
  display: inline-block;
}

/* Interactive Popover */
.smolitux-popover--interactive {
  pointer-events: auto;
}

.smolitux-popover--non-interactive {
  pointer-events: none;
}

/* Responsive Adjustments */
@media (max-width: 640px) {
  .smolitux-popover {
    max-width: calc(100vw - 2rem) !important;
  }
}

/* Dark Mode */
.dark .smolitux-popover {
  background-color: var(--popover-bg-dark, #1f2937);
  border-color: var(--popover-border-color-dark, #374151);
}

.dark .smolitux-popover-header {
  border-color: var(--popover-border-color-dark, #374151);
  color: var(--popover-header-color-dark, #f9fafb);
}

.dark .smolitux-popover-content {
  color: var(--popover-content-color-dark, #d1d5db);
}

.dark .smolitux-popover-arrow {
  background-color: var(--popover-bg-dark, #1f2937);
  border-color: var(--popover-border-color-dark, #374151);
}