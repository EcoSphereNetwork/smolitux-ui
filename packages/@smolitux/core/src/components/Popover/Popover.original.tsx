// packages/@smolitux/core/src/components/Popover/Popover.tsx
import React, { useState, useRef, useEffect } from 'react';

export type PopoverPlacement = 'top' | 'right' | 'bottom' | 'left' | 'top-start' | 'top-end' | 'right-start' | 'right-end' | 'bottom-start' | 'bottom-end' | 'left-start' | 'left-end';

export interface PopoverProps {
  /** Inhalt des Popovers */
  content: React.ReactNode;
  /** Trigger-Element */
  children: React.ReactElement;
  /** Position des Popovers */
  placement?: PopoverPlacement;
  /** Ist der Popover offen? (kontrollierter Modus) */
  isOpen?: boolean;
  /** Standard-Offenstatus (unkontrollierter Modus) */
  defaultOpen?: boolean;
  /** Callback beim Öffnen/Schließen */
  onOpenChange?: (isOpen: boolean) => void;
  /** Trigger-Ereignis */
  trigger?: 'click' | 'hover' | 'focus' | 'manual';
  /** Verzögerung vor dem Anzeigen (in ms) */
  openDelay?: number;
  /** Verzögerung vor dem Verstecken (in ms) */
  closeDelay?: number;
  /** Automatisch schließen, wenn außerhalb geklickt wird */
  closeOnClickOutside?: boolean;
  /** Automatisch schließen, wenn ESC gedrückt wird */
  closeOnEsc?: boolean;
  /** Pfeil anzeigen */
  showArrow?: boolean;
  /** Offset vom Trigger-Element (in px) */
  offset?: number;
  /** Maximale Breite des Popovers */
  maxWidth?: number | string;
  /** Titel des Popovers */
  title?: React.ReactNode;
  /** Zusätzliche CSS-Klassen */
  className?: string;
  /** z-Index für den Popover */
  zIndex?: number;
}

/**
 * Popover-Komponente für kontextuelle Informationen
 * 
 * @example
 * ```tsx
 * <Popover content="Details zu diesem Element">
 *   <Button>Mehr Info</Button>
 * </Popover>
 * ```
 */
export const Popover: React.FC<PopoverProps> = ({
  content,
  children,
  placement = 'bottom',
  isOpen: controlledIsOpen,
  defaultOpen = false,
  onOpenChange,
  trigger = 'click',
  openDelay = 0,
  closeDelay = 0,
  closeOnClickOutside = true,
  closeOnEsc = true,
  showArrow = true,
  offset = 8,
  maxWidth = 320,
  title,
  className = '',
  zIndex = 50,
}) => {
  // State für unkontrollierten Modus
  const [uncontrolledIsOpen, setUncontrolledIsOpen] = useState(defaultOpen);
  
  // Kontrollierter vs. unkontrollierter Modus
  const isControlled = controlledIsOpen !== undefined;
  const isOpen = isControlled ? controlledIsOpen : uncontrolledIsOpen;
  
  // Refs für DOM-Elemente
  const triggerRef = useRef<HTMLElement>(null);
  const popoverRef = useRef<HTMLDivElement>(null);
  
  // Refs für Timer
  const openTimerRef = useRef<NodeJS.Timeout | null>(null);
  const closeTimerRef = useRef<NodeJS.Timeout | null>(null);
  
  // Position des Popovers
  const [popoverPosition, setPopoverPosition] = useState({ top: 0, left: 0 });
  const [arrowPosition, setArrowPosition] = useState({ top: 0, left: 0 });
  
  // Helfer-Funktion zum Ändern des offenen Status
  const setOpen = (nextOpen: boolean) => {
    if (!isControlled) {
      setUncontrolledIsOpen(nextOpen);
    }
    if (onOpenChange) {
      onOpenChange(nextOpen);
    }
  };
  
  // Popover öffnen mit Verzögerung
  const openPopover = () => {
    if (closeTimerRef.current) {
      clearTimeout(closeTimerRef.current);
      closeTimerRef.current = null;
    }
    
    if (openDelay > 0) {
      openTimerRef.current = setTimeout(() => {
        setOpen(true);
      }, openDelay);
    } else {
      setOpen(true);
    }
  };
  
  // Popover schließen mit Verzögerung
  const closePopover = () => {
    if (openTimerRef.current) {
      clearTimeout(openTimerRef.current);
      openTimerRef.current = null;
    }
    
    if (closeDelay > 0) {
      closeTimerRef.current = setTimeout(() => {
        setOpen(false);
      }, closeDelay);
    } else {
      setOpen(false);
    }
  };
  
  // Popover-Status umschalten
  const togglePopover = () => {
    isOpen ? closePopover() : openPopover();
  };
  
  // Position des Popovers berechnen
  const calculatePosition = () => {
    if (!triggerRef.current || !popoverRef.current) return;
    
    const triggerRect = triggerRef.current.getBoundingClientRect();
    const popoverRect = popoverRef.current.getBoundingClientRect();
    const scrollTop = window.pageYOffset || document.documentElement.scrollTop;
    const scrollLeft = window.pageXOffset || document.documentElement.scrollLeft;
    
    // Viewport-Dimensionen
    const viewportWidth = window.innerWidth;
    const viewportHeight = window.innerHeight;
    
    // Standardposition und Pfeilposition
    let top = 0;
    let left = 0;
    let arrowTop = 0;
    let arrowLeft = 0;
    
    // Position basierend auf Placement berechnen
    switch (placement) {
      case 'top':
        top = triggerRect.top + scrollTop - popoverRect.height - offset;
        left = triggerRect.left + scrollLeft + (triggerRect.width / 2) - (popoverRect.width / 2);
        arrowTop = popoverRect.height;
        arrowLeft = popoverRect.width / 2;
        break;
      case 'top-start':
        top = triggerRect.top + scrollTop - popoverRect.height - offset;
        left = triggerRect.left + scrollLeft;
        arrowTop = popoverRect.height;
        arrowLeft = Math.min(triggerRect.width / 2, 20);
        break;
      case 'top-end':
        top = triggerRect.top + scrollTop - popoverRect.height - offset;
        left = triggerRect.right + scrollLeft - popoverRect.width;
        arrowTop = popoverRect.height;
        arrowLeft = popoverRect.width - Math.min(triggerRect.width / 2, 20);
        break;
      case 'right':
        top = triggerRect.top + scrollTop + (triggerRect.height / 2) - (popoverRect.height / 2);
        left = triggerRect.right + scrollLeft + offset;
        arrowTop = popoverRect.height / 2;
        arrowLeft = -5;
        break;
      case 'right-start':
        top = triggerRect.top + scrollTop;
        left = triggerRect.right + scrollLeft + offset;
        arrowTop = Math.min(triggerRect.height / 2, 20);
        arrowLeft = -5;
        break;
      case 'right-end':
        top = triggerRect.bottom + scrollTop - popoverRect.height;
        left = triggerRect.right + scrollLeft + offset;
        arrowTop = popoverRect.height - Math.min(triggerRect.height / 2, 20);
        arrowLeft = -5;
        break;
      case 'bottom':
        top = triggerRect.bottom + scrollTop + offset;
        left = triggerRect.left + scrollLeft + (triggerRect.width / 2) - (popoverRect.width / 2);
        arrowTop = -5;
        arrowLeft = popoverRect.width / 2;
        break;
      case 'bottom-start':
        top = triggerRect.bottom + scrollTop + offset;
        left = triggerRect.left + scrollLeft;
        arrowTop = -5;
        arrowLeft = Math.min(triggerRect.width / 2, 20);
        break;
      case 'bottom-end':
        top = triggerRect.bottom + scrollTop + offset;
        left = triggerRect.right + scrollLeft - popoverRect.width;
        arrowTop = -5;
        arrowLeft = popoverRect.width - Math.min(triggerRect.width / 2, 20);
        break;
      case 'left':
        top = triggerRect.top + scrollTop + (triggerRect.height / 2) - (popoverRect.height / 2);
        left = triggerRect.left + scrollLeft - popoverRect.width - offset;
        arrowTop = popoverRect.height / 2;
        arrowLeft = popoverRect.width;
        break;
      case 'left-start':
        top = triggerRect.top + scrollTop;
        left = triggerRect.left + scrollLeft - popoverRect.width - offset;
        arrowTop = Math.min(triggerRect.height / 2, 20);
        arrowLeft = popoverRect.width;
        break;
      case 'left-end':
        top = triggerRect.bottom + scrollTop - popoverRect.height;
        left = triggerRect.left + scrollLeft - popoverRect.width - offset;
        arrowTop = popoverRect.height - Math.min(triggerRect.height / 2, 20);
        arrowLeft = popoverRect.width;
        break;
      default:
        break;
    }
    
    // Position innerhalb des Viewports halten
    if (left < 10) {
      arrowLeft += left - 10;
      left = 10;
    } else if (left + popoverRect.width > viewportWidth - 10) {
      const overflow = (left + popoverRect.width) - (viewportWidth - 10);
      arrowLeft += overflow;
      left -= overflow;
    }
    
    if (top < 10) {
      arrowTop += top - 10;
      top = 10;
    } else if (top + popoverRect.height > viewportHeight + scrollTop - 10) {
      // Wenn zu weit unten, oberhalb des Triggers anzeigen
      if (placement.startsWith('bottom')) {
        top = triggerRect.top + scrollTop - popoverRect.height - offset;
        arrowTop = popoverRect.height;
      } else {
        const overflow = (top + popoverRect.height) - (viewportHeight + scrollTop - 10);
        arrowTop += overflow;
        top -= overflow;
      }
    }
    
    setPopoverPosition({ top, left });
    setArrowPosition({ top: arrowTop, left: arrowLeft });
  };
  
  // ESC-Taste zum Schließen behandeln
  useEffect(() => {
    if (!closeOnEsc) return;
    
    const handleEscape = (e: KeyboardEvent) => {
      if (e.key === 'Escape' && isOpen) {
        closePopover();
      }
    };
    
    document.addEventListener('keydown', handleEscape);
    return () => document.removeEventListener('keydown', handleEscape);
  }, [isOpen, closeOnEsc]);
  
  // Klick außerhalb zum Schließen
  useEffect(() => {
    if (!closeOnClickOutside || !isOpen) return;
    
    const handleClick = (e: MouseEvent) => {
      if (
        popoverRef.current && 
        triggerRef.current &&
        !popoverRef.current.contains(e.target as Node) &&
        !triggerRef.current.contains(e.target as Node)
      ) {
        closePopover();
      }
    };
    
    document.addEventListener('mousedown', handleClick);
    return () => document.removeEventListener('mousedown', handleClick);
  }, [isOpen, closeOnClickOutside]);
  
  // Neuberechnung der Position, wenn sich die Größe ändert
  useEffect(() => {
    if (!isOpen) return;
    
    calculatePosition();
    
    const handleResize = () => calculatePosition();
    window.addEventListener('resize', handleResize);
    window.addEventListener('scroll', handleResize, true);
    
    return () => {
      window.removeEventListener('resize', handleResize);
      window.removeEventListener('scroll', handleResize, true);
    };
  }, [isOpen]);
  
  // Aufräumen der Timer
  useEffect(() => {
    return () => {
      if (openTimerRef.current) clearTimeout(openTimerRef.current);
      if (closeTimerRef.current) clearTimeout(closeTimerRef.current);
    };
  }, []);
  
  // Position nach der ersten Renderung berechnen
  useEffect(() => {
    if (isOpen) {
      requestAnimationFrame(calculatePosition);
    }
  }, [isOpen]);
  
  // Events für den Trigger basierend auf dem Trigger-Typ
  const getTriggerProps = () => {
    const commonProps = {
      ref: (el: Element | null) => {
        // Ref vom Original beibehalten, wenn vorhanden
        const child = React.Children.only(children);
        if (React.isValidElement(child)) {
          const originalRef = (child as any).ref;
          if (originalRef && typeof originalRef === 'function') {
            originalRef(el);
          } else if (originalRef && typeof originalRef === 'object' && 'current' in originalRef) {
            (originalRef as React.MutableRefObject<HTMLElement | null>).current = el as HTMLElement;
          }
        }
        if (triggerRef) {
          (triggerRef as React.MutableRefObject<HTMLElement | null>).current = el as HTMLElement;
        }
      }
    };
    
    if (trigger === 'manual') {
      return commonProps;
    }
    
    if (trigger === 'click') {
      return {
        ...commonProps,
        onClick: (e: React.MouseEvent) => {
          togglePopover();
          if (children.props.onClick) {
            children.props.onClick(e);
          }
        }
      };
    }
    
    if (trigger === 'hover') {
      return {
        ...commonProps,
        onMouseEnter: (e: React.MouseEvent) => {
          openPopover();
          if (children.props.onMouseEnter) {
            children.props.onMouseEnter(e);
          }
        },
        onMouseLeave: (e: React.MouseEvent) => {
          closePopover();
          if (children.props.onMouseLeave) {
            children.props.onMouseLeave(e);
          }
        }
      };
    }
    
    if (trigger === 'focus') {
      return {
        ...commonProps,
        onFocus: (e: React.FocusEvent) => {
          openPopover();
          if (children.props.onFocus) {
            children.props.onFocus(e);
          }
        },
        onBlur: (e: React.FocusEvent) => {
          closePopover();
          if (children.props.onBlur) {
            children.props.onBlur(e);
          }
        }
      };
    }
    
    return commonProps;
  };
  
  // Pfeilform basierend auf Placement
  const getArrowStyles = () => {
    // Basis-Styles
    const baseStyles: React.CSSProperties = {
      position: 'absolute',
      width: '10px',
      height: '10px',
      transform: 'rotate(45deg)',
      backgroundColor: 'inherit',
      border: 'inherit',
      zIndex: -1
    };
    
    if (placement.startsWith('top')) {
      return {
        ...baseStyles,
        bottom: '-5px',
        borderBottom: '1px solid #e2e8f0',
        borderRight: '1px solid #e2e8f0',
        borderTop: 'none',
        borderLeft: 'none'
      };
    } else if (placement.startsWith('right')) {
      return {
        ...baseStyles,
        left: '-5px',
        borderLeft: '1px solid #e2e8f0',
        borderBottom: '1px solid #e2e8f0',
        borderTop: 'none',
        borderRight: 'none'
      };
    } else if (placement.startsWith('bottom')) {
      return {
        ...baseStyles,
        top: '-5px',
        borderTop: '1px solid #e2e8f0',
        borderLeft: '1px solid #e2e8f0',
        borderBottom: 'none',
        borderRight: 'none'
      };
    } else if (placement.startsWith('left')) {
      return {
        ...baseStyles,
        right: '-5px',
        borderRight: '1px solid #e2e8f0',
        borderTop: '1px solid #e2e8f0',
        borderBottom: 'none',
        borderLeft: 'none'
      };
    }
    
    return baseStyles;
  };
  
  // Trigger Element mit neuen Props klonen
  const triggerElement = React.cloneElement(
    React.Children.only(children),
    getTriggerProps()
  );
  
  return (
    <>
      {triggerElement}
      
      {isOpen && (
        <div
          ref={popoverRef}
          role="tooltip"
          className={`
            absolute z-${zIndex} bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            ${className}
          `}
          style={{
            top: popoverPosition.top,
            left: popoverPosition.left,
            maxWidth: typeof maxWidth === 'number' ? `${maxWidth}px` : maxWidth
          }}
        >
          {/* Title */}
          {title && (
            <div className="mb-2 font-medium text-gray-900 dark:text-white border-b border-gray-200 dark:border-gray-700 pb-2">
              {title}
            </div>
          )}
          
          {/* Content */}
          <div>{content}</div>
          
          {/* Arrow */}
          {showArrow && (
            <div
              style={{
                ...getArrowStyles(),
                top: arrowPosition.top === 0 ? undefined : arrowPosition.top,
                left: arrowPosition.left === 0 ? undefined : arrowPosition.left
              }}
            />
          )}
        </div>
      )}
    </>
  );
};

export default Popover;
