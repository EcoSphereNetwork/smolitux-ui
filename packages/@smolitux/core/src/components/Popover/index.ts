// packages/@smolitux/core/src/components/Popover/index.ts
export { default, type PopoverProps, type PopoverPlacement } from './Popover';
