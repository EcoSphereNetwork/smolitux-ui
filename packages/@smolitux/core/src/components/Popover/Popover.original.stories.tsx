import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import { Popover.original } from './Popover.original';

const meta: Meta<typeof Popover.original> = {
  title: 'Components/Popover.original',
  component: Popover.original,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Popover.original',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Popover.original',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Popover.original',
    onClick: () => alert('Clicked!'),
  },
};
