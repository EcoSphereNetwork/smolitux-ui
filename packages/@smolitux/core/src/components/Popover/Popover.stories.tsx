import type { Meta, StoryObj } from '@storybook/react';
import { Popover } from './Popover';

const meta: Meta<typeof Popover> = {
  title: 'Components/Popover',
  component: Popover,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Popover',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Popover',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Popover',
    onClick: () => alert('Clicked!'),
  },
};

export const WithCloseButton: Story = {
  args: {
    content: 'Closable content',
    showCloseButton: true,
    children: <button>Trigger</button>,
  },
};
