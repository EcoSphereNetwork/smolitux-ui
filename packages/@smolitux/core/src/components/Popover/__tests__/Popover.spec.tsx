import React from 'react';
import { render } from '@testing-library/react';
import { Popover } from '../Popover';

describe('Popover Snapshots', () => {
  it('renders default popover correctly', () => {
    const { asFragment } = render(
      <Popover content="Popover Content" isOpen={true}>
        <button>Trigger</button>
      </Popover>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with arrow correctly', () => {
    const { asFragment } = render(
      <Popover 
        content="Popover Content" 
        isOpen={true}
        showArrow={true}
      >
        <button>Trigger</button>
      </Popover>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with different placements correctly', () => {
    const placements: Array<'top' | 'right' | 'bottom' | 'left'> = ['top', 'right', 'bottom', 'left'];
    
    const fragments = placements.map(placement => {
      const { asFragment } = render(
        <Popover 
          content="Popover Content" 
          isOpen={true}
          placement={placement}
        >
          <button>Trigger</button>
        </Popover>
      );
      return { placement, fragment: asFragment() };
    });
    
    fragments.forEach(({ placement, fragment }) => {
      expect(fragment).toMatchSnapshot(`Popover with placement ${placement}`);
    });
  });

  it('renders with rich content correctly', () => {
    const { asFragment } = render(
      <Popover 
        content={
          <div>
            <h3>Popover Title</h3>
            <p>This is a paragraph of text.</p>
            <button>Action</button>
          </div>
        } 
        isOpen={true}
      >
        <button>Trigger</button>
      </Popover>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom width correctly', () => {
    const { asFragment } = render(
      <Popover 
        content="Popover Content" 
        isOpen={true}
        width="300px"
      >
        <button>Trigger</button>
      </Popover>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom className correctly', () => {
    const { asFragment } = render(
      <Popover 
        content="Popover Content" 
        isOpen={true}
        className="custom-popover"
      >
        <button>Trigger</button>
      </Popover>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});