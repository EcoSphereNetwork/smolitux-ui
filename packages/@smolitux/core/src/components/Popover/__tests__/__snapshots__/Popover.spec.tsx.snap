// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Snapshots renders default popover correctly 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; border-top: 1px solid #e2e8f0; border-left: 1px solid #e2e8f0; top: -7px; left: -10px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with arrow correctly 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; border-top: 1px solid #e2e8f0; border-left: 1px solid #e2e8f0; top: -7px; left: -10px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with custom className correctly 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            custom-popover
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; border-top: 1px solid #e2e8f0; border-left: 1px solid #e2e8f0; top: -7px; left: -10px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with custom width correctly 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; border-top: 1px solid #e2e8f0; border-left: 1px solid #e2e8f0; top: -7px; left: -10px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with different placements correctly: Popover with placement bottom 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; border-top: 1px solid #e2e8f0; border-left: 1px solid #e2e8f0; top: -7px; left: -10px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with different placements correctly: Popover with placement left 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; right: -5px; border-right: 1px solid #e2e8f0; border-top: 1px solid #e2e8f0; top: -10px; left: -18px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with different placements correctly: Popover with placement right 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; border-left: 1px solid #e2e8f0; border-bottom: 1px solid #e2e8f0; left: -7px; top: -10px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with different placements correctly: Popover with placement top 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      Popover Content
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; bottom: -5px; border-bottom: 1px solid #e2e8f0; border-right: 1px solid #e2e8f0; top: -18px; left: -10px;"
    />
  </div>
</DocumentFragment>
`;

exports[`Popover Snapshots renders with rich content correctly 1`] = `
<DocumentFragment>
  <button>
    Trigger
  </button>
  <div
    class="
            absolute z-50 bg-white dark:bg-gray-800 
            border border-gray-200 dark:border-gray-700 
            rounded-md shadow-md p-4
            
          "
    role="tooltip"
    style="top: 10px; left: 10px; max-width: 320px;"
  >
    <div>
      <div>
        <h3>
          Popover Title
        </h3>
        <p>
          This is a paragraph of text.
        </p>
        <button>
          Action
        </button>
      </div>
    </div>
    <div
      style="position: absolute; width: 10px; height: 10px; transform: rotate(45deg); background-color: inherit; z-index: -1; border-top: 1px solid #e2e8f0; border-left: 1px solid #e2e8f0; top: -7px; left: -10px;"
    />
  </div>
</DocumentFragment>
`;
